js/wavesexchange.js

```diff
@@ -1234,6 +1237,40 @@ module.exports = class wavesexchange extends Exchange {
         };
     }
 
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        this.checkRequiredDependencies ();
+        this.checkRequiredKeys ();
+        await this.loadMarkets ();
+        let market = undefined;
+        if (symbol !== undefined) {
+            market = this.market (symbol);
+        }
+        const timestamp = this.milliseconds ();
+        const byteArray = [
+            this.base58ToBinary (this.apiKey),
+            this.numberToBE (timestamp, 8),
+        ];
+        const binary = this.binaryConcatArray (byteArray);
+        const hexSecret = this.binaryToBase16 (this.base58ToBinary (this.secret));
+        const signature = this.eddsa (this.binaryToBase16 (binary), hexSecret, 'ed25519');
+        const request = {
+            'Timestamp': timestamp.toString (),
+            'Signature': signature,
+            'publicKey': this.apiKey,
+            'orderId': id,
+        };
+        const response = await this.matcherGetMatcherOrderbookPublicKeyOrderId (this.extend (request, params));
+        return this.parseOrder (response, market);
+    }
+
+    async fetchOpenOrder (id, symbol = undefined, params = {}) {
+        return this.fetchOrder (id, symbol, params);
+    }
+
+    async fetchClosedOrder (id, symbol = undefined, params = {}) {
+        return this.fetchOrder (id, symbol, params);
+    }
```

Comment:
I would remove fetchClosedOrder and fetchOpenOrder, usually we don't emulate them if there is fetchOrder available.